src/lib/ng-package/entry-point/write-package.transform.ts

```diff
@@ -35,55 +35,81 @@ export const writePackageTransform = (options: NgPackagrOptions) =>
     ];
 
     if (!ngEntryPoint.isSecondaryEntryPoint) {
-      const assetFiles: string[] = [];
-
       // COPY ASSET FILES TO DESTINATION
       spinner.start('Copying assets');
 
       try {
-        for (const asset of [...ngPackage.assets, 'LICENSE', '**/README.md']) {
-          let assetFullPath = path.join(ngPackage.src, asset);
-
+        type AssetCopyTask = [srcPath: string, destPath: string];
+        const assetCopyTasks: AssetCopyTask[] = [];
+
+        // Parse string assets and add to tasks
+        const assetGlobs = ngPackage.assets
+          .filter((asset): asset is string => typeof asset === 'string')
+          .concat('LICENSE', '**/README.md');
+        for (const assetGlob of assetGlobs) {
+          const assetFullGlob = path.join(ngPackage.src, assetGlob);
+          let assetFullGlobIs: 'file' | 'dir' | null = null;
           try {
-            const stats = await stat(assetFullPath);
-            if (stats.isFile()) {
-              assetFiles.push(assetFullPath);
-              continue;
-            }
-
-            if (stats.isDirectory()) {
-              assetFullPath = path.join(assetFullPath, '**/*');
-            }
+            const stats = await stat(assetFullGlob);
+            const isFile = stats.isFile() || assetGlob === 'LICENSE';
+            const isDir = stats.isDirectory();
+            assetFullGlobIs = isFile ? 'file' : isDir ? 'dir' : null;
           } catch {}
 
-          if (asset === 'LICENSE') {
-            continue;
+          const assetFilePaths: string[] = [];
+          if (assetFullGlobIs == 'file') {
+            assetFilePaths.push(assetFullGlob);
+          } else {
+            const glob = assetFullGlobIs == 'dir' ? path.join(assetFullGlob, '**/*') : assetFullGlob;
+            const paths = await globFiles(glob, {
+              ignore: ignorePaths,
+              cache: ngPackageNode.cache.globCache,
+              dot: true,
+              nodir: true,
+            });
+            assetFilePaths.push(...paths);
           }
 
-          const files = await globFiles(assetFullPath, {
-            ignore: ignorePaths,
+          const tasks: AssetCopyTask[] = assetFilePaths.map(filePath => [
+            filePath,
+            path.resolve(ngPackage.dest, path.relative(ngPackage.src, filePath)),
+          ]);
+          assetCopyTasks.push(...tasks);
+        }
+
+        // Parse object assets and add to tasks
+        const assetEntries = ngPackage.assets.filter((item): item is AssetEntry => typeof item !== 'string');
+        for (const assetEntry of assetEntries) {
+          const srcDirAbsPath = path.join(ngPackage.src, assetEntry.input);
+          const destDirAbsPath = path.join(ngPackage.dest, assetEntry.output);
```

Comment:
Since now the output can be provided, we need to valid that this doesn't escape the `dist`.